File: OraculumApi/Models/FrontOffice/Message.cs

namespace OraculumApi.Models.FrontOffice
{
    public partial class Message : WeaviateEntity, IEntity<MessageDTO>
    {
        public string? sender;
        public string? text;
        public string[]? factIds;
        public string[]? extraFactIds;
        public DateTime? timestamp;

        public MessageDTO toDTO()
        {
            return new MessageDTO()
            {
                Id = this.id,
                Sender = this.sender,
                Text = this.text,
                FactIds = this.factIds,
                ExtraFactIds = this.extraFactIds,
                Timestamp = this.timestamp
            };
        }
    }
}

File: web/OraculumFE/src/app/interfaces/message.ts
import { SafeHtml } from '@angular/platform-browser';

export interface Message {
  id: string;
  text: any;
  sender: string;
  timestamp: string;
  extraFactIds?: any;
  factIds?: any;
  feedback?: boolean;
  feedbackText?: string;
  completed?: boolean;
  url?: string;
  sources?: { url: string; display: string }[]; // Add sources array to store URLs
}
